include/open62541/plugin/accesscontrol.h

```diff
@@ -16,6 +16,34 @@ _UA_BEGIN_DECLS
 struct UA_AccessControl;
 typedef struct UA_AccessControl UA_AccessControl;
 
+#ifdef UA_ENABLE_ROLE_PERMISSION
+
+typedef enum {
+    UA_ANONYMOUS_WELL_KNOWN_RULE = 0,
+    UA_AUTHENTICATEDUSER_WELL_KNOWN_RULE = 1,
+    UA_CONFIGUREADMIN_WELL_KNOWN_RULE = 2,
+    UA_ENGINEER_WELL_KNOWN_RULE = 3,
+    UA_OBSERVER_WELL_KNOWN_RULE = 4,
+    UA_OPERATOR_WELL_KNOWN_RULE = 5,
+    UA_SECURITYADMIN_WELL_KNOWN_RULE = 6,
+    UA_SUPERVISOR_WELL_KNOWN_RULE = 7,
+}UA_AccessControlGroup;
```

Comment:
1. Please include all well known roles: https://reference.opcfoundation.org/Core/Part3/v105/docs/4.8
2. The following naming-scheme is preferred: `UA_ROLE_ANONYMOUS` and similar.
3. AccessControlGroup is a funky name for OPC UA Roles. Rename this to `UA_Role`.
4. Add a (const) array of NodeId called `UA_ROLES` where the Role-index maps to the NodeId of the role.
    Note that users can add their own roles. These will not show up in the enum or the NodeId array.
    So the Roles-Enum is only a shortcut. We should always use the NodeId role identifier to allow the extension with custom roles going forward.